Code Review

Merdoth opened the pull request:
#### What does this PR do?

- N/A

#### Description of Task to be completed?

- N/A

#### How should this be manually tested?

- run ```git clone``` this repository
- navigate to the project directory
- run ```npm install```
- run ```npm run server``` from the command line
- sign up as a user
- navigate to profile
- click on manage account 
- click on add recipes

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
- story type: N/A
- story title: N/A

#### Screenshots (if appropriate)
N/A

#### Question
N/A

Merdoth committed:
Restructes Project Folder and Updates profile,recipes,style

Merdoth committed:
Modifies index.html

Merdoth committed:
Merge pull request #1 from Merdoth/template

Modifies index.html

Merdoth committed:
Initial commit

Merdoth committed:
Modifies server

Merdoth committed:
Setups server

Merdoth committed:
Modifies router

Merdoth committed:
Adds the modifications

Merdoth committed:
Restructs Server

Merdoth committed:
Merge pull request #2 from Merdoth/server-setup

Server setup

Merdoth committed:
Modifies code

Merdoth committed:
Restructs code

Merdoth committed:
Adds migrations,models and configuration

Merdoth committed:
Updates server

Merdoth committed:
add gitignore,sequelizerc,travis and package.json

Merdoth committed:
Merge branch 'develop' into server-setup

Merdoth committed:
Merge pull request #3 from Merdoth/server-setup

Server setup

Merdoth committed:
Modifies server: controller, config, models, routes

Merdoth committed:
Merge pull request #4 from Merdoth/ft-create-users-account-151661025

Modifies server: controller, config, models, routes

Merdoth committed:
Restructures recipes.js,user.js,users.js and token.js

Merdoth committed:
Merge pull request #5 from Merdoth/ft-login-users-account-151661097

Restructures recipes.js,user.js,users.js and token.js

Merdoth committed:
Adds changes implemented in server

Merdoth committed:
Merge pull request #6 from Merdoth/ft-add-recipe-151661119

Adds changes implemented in server

Merdoth committed:
get recipe route and get recipe controller

Merdoth committed:
Merge pull request #7 from Merdoth/ft-get-all-recipes-151661167

 get recipe route and get recipe controller

Merdoth committed:
Modifies server and unstaged env

Merdoth committed:
Merge pull request #8 from Merdoth/ft-modify-recipe-151661142

Modifies server and unstaged env

Merdoth committed:
Implements changes in code

Merdoth committed:
Merge pull request #9 from Merdoth/ft-create-user-151661025

Implements changes in code

Merdoth committed:
Modifies package.json

Merdoth committed:
Adds sequelize-cli

Merdoth committed:
Modifies code

Merdoth committed:
specifies installations for travis

Merdoth committed:
installs dependencies and runs it

Merdoth committed:
Restructes travis and package.json

Merdoth committed:
Update README.md

Merdoth committed:
Update README.md

Merdoth committed:
Update README.md

Merdoth committed:
Restructes code

Merdoth committed:
chore: fix merge conflict

Merdoth committed:
feature: restructes controllers,package.json,models,routes and utils

Merdoth committed:
feature: fixes server(review,recipes and votes controllers,models and routes)

Merdoth committed:
feature: Adds favorites and modifes server

Merdoth committed:
feature: Reconstructs server and adds middleware

Merdoth committed:
Merge pull request #10 from Merdoth/ft-votes-favorites-update

Ft votes favorites update

Merdoth committed:
task:  Sets up react and redux in client folder

Merdoth committed:
Task: Sets up react, reactdom, webpack,etc

Merdoth committed:
Feature: Adds node version

Merdoth committed:
Feature: Renames babel to babel-cli in scripts bulid

Merdoth committed:
Feature: Adds changes to script

Merdoth committed:
Chore: Render main react component

Merdoth committed:
Chore: setsup client-side, modifies package.json.babelrc and webpack Config

Merdoth committed:
Feature: Adds reacct components to the client side, Chore: Setup main.scss,image,node-sass

Merdoth committed:
Feature: Changes template homepage features into react components

Merdoth committed:
Implements react changes on both client side and server side

Merdoth committed:
Feature: Validates server side user and client side user

Merdoth committed:
Feature: Updates and connects but cilent-side and server-side

Merdoth committed:
Feature: Redirects on successful server response

Merdoth committed:
Feature: Adds flash message reducer

Merdoth committed:
Feature: Renders flash messages and allows to delete them

Commit:
Heals package.json script build error and runs sequlize migrations

Commit:
Modifies application

Merdoth committed:
Add Heroku generated app.json

Commit:
Update app.json

Commit:
Move babel from devDependencies to dependencies to fix heroku build

Commit:
Modifies code

Commit:
Changes dependencies

Commit:
Changes dependencies

Commit:
feat: change the naming conventions

Commit:
feat: change the naming conventions

Commit:
change route

Merdoth committed:
Merge pull request #11 from Merdoth/ft-client-side

Ft client side

Commit:
feat(sigup): implement user signup validtion _

Merdoth committed:
Merge pull request #12 from Merdoth/ft-user-signup-152849897

feat(sigup): implement user signup validation

Commit:
route-ish

Commit:
fix: fix merge conflict

Commit:
feat(signin):  Implement user signin validation
-refractor signup route(bug fixing)

Merdoth committed:
Merge pull request #13 from Merdoth/ft-user-signin-152852746

feat(signin):  Implement user signin validation

Commit:
feat(createRecipeValidator): checks for the validity of what the user is passing
- checks if a user is passing in an empty space or multiple empty spaces
[Delievers #152873614]

Merdoth committed:
Merge pull request #14 from Merdoth/ft-create-recipe-152873614

feat(createRecipeValidator): checks for the validity of what the user…

Commit:
feat(getAllMyRecipes): adds a function that allows a user get all his/her recipes.
- enabled route for user getAllMy recipes

Commit:
fix(bug): just a little change

Merdoth committed:
Merge pull request #15 from Merdoth/ft-gellAllMy-recipes-152874095

Ft get all my recipes 152874095

Commit:
fix(bug): follows airbnb configuration for es-lint
- corrects code to adjust to these rules in the configuration
[Delievers#152889620]

Merdoth committed:
Merge pull request #16 from Merdoth/fix-eslint-152889620

fix(bug): follows airbnb configuration for es-lint

Commit:
feat(user test): tested for all user edgecases;
- made sure what was runing on validator is the same as the one on the test
- modify my package.json
- updates coverage

Merdoth committed:
Merge pull request #17 from Merdoth/ft-user-test-152919448

feat(user test): tested for all user edgecases;

Commit:
feature(recipe-test): add recipe test
- update middlewares
- restruct controllers
- updates package.json
Delievers(#152999106)

Merdoth committed:
fix(bug): update server controllers and client components

Merdoth committed:
feature: update packages and reconstruct app

Merdoth committed:
update changes in client folder and server folder

Merdoth committed:
feature(protect-page): update changes in client folder.
- update webpack, package.json and babelrc

Merdoth committed:
feature(protect-page): - update package.json and webpack
- rename files and reconstruct code structuring
- update and add new components
- restyle component design
-Add authentication and middleware to both server and client sides

[Delivers #154119022]

Merdoth committed:
Merge pull request #18 from Merdoth/ft-protect-page-154119022

Ft protect page 154119022

Merdoth committed:
feature(user-recipes): - add recipe component
- update index.js route to include recipe route
- update scss to fix recipe and menu styles
- update client components

[Delivers #154282765]

Merdoth committed:
Merge pull request #19 from Merdoth/feature/154282765/user-recipes

feature(user-recipes): - add recipe component

Merdoth committed:
feature(add-recipes): - update client index.js to accommodate  addrecipes route
- rename topRecipes reducer to recipes reducer in rootReducer.js
-  update actions to accommodate recipe success and recipe failure
- add addrecipe component to client components
- add recipe reducers to client reducers
- restyle scss for drop down
- add addRecipeRequest to api calls
- update server recipe controller to accommodate image upload
-  update server middleware, migrations and models to accommodate image entry field.

[Delivers #154307102]

Merdoth committed:
Merge pull request #20 from Merdoth/feature/154307102/add-recipes

feature(add-recipes): - update client index.js to accommodate  addrec…

Merdoth committed:
feature(populate-recipe-page): - update recipe components to display recipes from database
- update reducers and morerecipes api

Merdoth committed:
feature(populate-recipe-page): -update recipe components to fix bug

[Delivers #154341872]

Merdoth committed:
Merge pull request #21 from Merdoth/feature/154341872/populate-recipe-page

Feature/154341872/populate recipe page

Merdoth committed:
trying todo

Merdoth committed:
feature(display-recipe-details): - add reciple-details component
- link  actions to recipe-details
- link reducers to recipe-details
- update travis.yml file

Merdoth committed:
just commit

Merdoth committed:
feature(dispiay-recipe-details): - add update recipe component so a user can be able to edit and update his/her recipes
- add recipe details component so a user can be able to view his or her added recipes

[Delivers #154379126]

Merdoth committed:
Merge pull request #23 from Merdoth/feature/154379126/display-recipe-details

Feature/154379126/display recipe details

nebanat commented on .eslintrc:
```diff
@@ -0,0 +1,70 @@
+    ],
+    "react/jsx-uses-react": "error",
+    "react/jsx-uses-vars": "error",
+    // "valid-jsdoc": ["error", {
```
remove all commented codes

Merdoth replied:
resolved

Merdoth replied:
Done

nebanat commented on client/components/Footer.jsx:
```diff
@@ -0,0 +1,18 @@
+    <div className="container-fluid">
+      <div className="footer">
+        <div>
+          <p><i>Designed with</i> <i className="fa fa-heart love" aria-hidden="true"></i> <i>by </i><strong>Ucheya</strong></p>
```
Lines of code should not exceed 80 characters

Merdoth replied:
Done, except the code lines not flagged by eslint

nebanat commented on client/components/NavigationBar.jsx:
```diff
@@ -0,0 +1,152 @@
+            </a>
+            <a
+              className="dropdown-item"
+              style={{ color: '#000' }}
```
remove all inline stylings...have a seperate .css file

nebanat commented on client/components/Profile.jsx:
```diff
@@ -0,0 +1,119 @@
+    event.preventDefault();
+    this.setState({ errors: {}, isLoading: true });
+    // this.props
+    //   .userSigninRequest(this.state)
```
Again remove all commented code

Merdoth replied:
Done

nebanat commented on client/components/Recipes/RecipeCard/RecipeCard.jsx:
```diff
@@ -0,0 +1,42 @@
+import image from '../../../image/Edika-Ikong.jpg';
+
+// const title = 'Edika Ikong';
+// const text = `Cook the spaghetti in salted water about half of the way
```
Again remove all commented out code

Merdoth replied:
Done

nebanat commented on client/components/Recipes/RecipeCard/RecipeCard.jsx:
```diff
@@ -0,0 +1,42 @@
+import RecipeCardImage from './RecipeCardImage.jsx';
+import RecipeCardDes from './RecipeCardDes.jsx';
+import RecipeCardFooter from './RecipeCardFooter.jsx';
+import image from '../../../image/Edika-Ikong.jpg';
```
Why are you importing jpg files ?

Merdoth replied:
Forgot to remove it after I implemented cloudinary upload access, but it's been removed now

nebanat commented on client/components/Recipes/RecipeDetails.jsx:
```diff
@@ -0,0 +1,103 @@
+import { getOneRecipe } from '../../actions/recipeActions/recipeActions';
+import { Icons } from '../common/Icons.jsx';
+import RecipeCardImage from './RecipeCard/RecipeCardImage.jsx';
+import RecipeCardFooter1 from './RecipeCard/RecipeCardFooter1.jsx';
```
RecipeFooter1???Use more approipriate variable naming ....RecipeFooter will be fine

nebanat commented on client/components/common/TextArea.jsx:
```diff
@@ -0,0 +1,24 @@
+      onChange={onChange}
+      className="form-control"
+    />
+    <br />
```
avoid using breaks, better off applying a margin

nebanat commented on client/index.js:
```diff
@@ -0,0 +1,70 @@
+if (jwtToken) {
+  const valid = jwt.verify(
+    jwtToken,
+    'this23423girl223is#$3423crazy',
```
this keys should be moved to your .env file

nebanat commented on client/rootReducer.js:
```diff
@@ -0,0 +1,15 @@
+import { combineReducers } from 'redux';
+
+import recipes from './reducers/recipeReducer';
```
default imports should match the base filename e.g.  import `recipeReducer` from './reducers/`recipeReducer`'

Merdoth committed:
feature(delete-recipe and refractor code):
- separate webpack development configuration from webpack production configuration
- remove all commented code
- add JSDoc to every class and method
- run migrations for server models
- update rootReducer to accommodate  necessary imports
- remove all unneeded console.log messages
- removed mocha and replaced with an active version
- removed flash message folders
- add delete action
- modified  Navigation Bar
- updated travis.yml file

[Delivers #154507643]

Merdoth committed:
feature(delete-recipe and refractor code):
- separate webpack development configuration from webpack production configuration
- remove all commented code
- add JSDoc to every class and method
- run migrations for server models
- update rootReducer to accommodate  necessary imports
- remove all unneeded console.log messages
- removed mocha and replaced with an active version
- removed flash message folders
- add delete action
- modified  Navigation Bar
- updated travis.yml file

[Delivers #154507643]

emekanwankwo commented on coverage/base.css:
```diff
@@ -0,0 +1,212 @@
+body, html {
```
I don't think you need the coverage folder on github, you should consider adding it to `.gitignore`

emekanwankwo commented on .sequelizerc:
```diff
@@ -0,0 +1,9 @@
+const path = require('path');
+// import path from 'path';
```
Remove commented code as already pointed out

andela-sjames commented on client/actions/auth/authActions.js:
```diff
@@ -0,0 +1,75 @@
+ */
+export function logout() {
+  return (dispatch) => {
+    localStorage.removeItem('jwtToken');
```
Consider using redux middleware for local storage actions

emekanwankwo commented on README.md:
```diff
@@ -1,2 +1,53 @@
+[![Coverage Status](https://coveralls.io/repos/github/Merdoth/more-recipes/badge.svg?branch=develop)](https://coveralls.io/github/Merdoth/more-recipes?branch=develop)
+
+
+More recipes is a web appliaction that allows food lovers to exhibit and explore their ideas towards food, and gets the chance to share with others what they've been able to come across in the course of exploring. It features are:
```
`Nitpick`: I believe the last sentence should've been `Its features are: . . .` or `It features: . . .`

andela-sjames commented on client/components/Recipes/AddRecipe/AddRecipePage.jsx:
```diff
@@ -0,0 +1,19 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import AddRecipeForm from './AddRecipeForm.jsx';
```
You don't have to append the extension name `.jsx` to the file(s) you are importing...

andela-sjames commented on client/components/Recipes/RecipeCard/RecipeCard.jsx:
```diff
@@ -0,0 +1,42 @@
+   * @memberof ReciprecipeeCard
+   */
+  render() {
+    console.log('this is very expected ', this.props.recipeList);
```
Please remove `console.log()`

andela-sjames commented on client/components/Recipes/RecipeDetails.jsx:
```diff
@@ -0,0 +1,103 @@
+  }
+  render() {
+    const recipeDetails = this.state.recipes;
+    console.log('recioe details', recipeDetails);
```
Console.log()?

andela-sjames commented on client/components/Recipes/Recipes.jsx:
```diff
@@ -0,0 +1,53 @@
+      <div>
+        <div className="container manage">
+          <h2>Recipes</h2>
+          <p>Feel free to manipulate your own account</p>
```
Manipulate my account? not sure how this sounds though

emekanwankwo commented on client/components/Home.jsx:
```diff
@@ -0,0 +1,52 @@
+import ProptTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import Slider from './Slider.jsx';
```
`Nitpick`: It'll be a lot cleaner to maintain consistency with adding or not adding file extensions (`.jsx`) to import statements.

andela-sjames commented on client/components/Signin/SigninForm.jsx:
```diff
@@ -0,0 +1,154 @@
+  }
+
+  /**
+   *  this method is gets the values of the input
```
Want to consider the words used here though

andela-sjames commented on client/components/Signin/SigninForm.jsx:
```diff
@@ -0,0 +1,154 @@
+  }
+
+  /**
+   * this method handle errors that return as an object
```
errors that return as an object?

emekanwankwo commented on client/components/Profile.jsx:
```diff
@@ -0,0 +1,119 @@
+    this.state = {
+      fName: '',
+      eName: '',
+      cName: '',
```
You should consider making variable names a bit more descriptive.

emekanwankwo commented on client/components/Recipes/RecipeCard/RecipeCardFooter1.jsx:
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+// import Button from '../../common/Button.jsx';
```
Remove commented code as already pointed out

Merdoth committed:
running travis

femisenjobi commented on client/actions/auth/signupActions.js:
```diff
@@ -0,0 +1,14 @@
+import * as types from '../actionTypes';
```
It is good practice to have your action files by resources (eg user) as opposed to actions (signin and signout actions). a `userActions` with signin and signout feels more appropriate.

This comment goes for flashMessages too. Should not be add and delete in seperate files

femisenjobi commented on client/components/NavigationBar.jsx:
```diff
@@ -0,0 +1,152 @@
+        <li>
+          <Link className="nav-link" to="/recipes">
+            <i className="fa fa-home" aria-hidden="true">
+              {' '}
```
I'm assuming the purpose of this is to create some form of padding for the content. You can achieve this with styling. Padding top or left will save you this.

femisenjobi commented on client/components/Profile.jsx:
```diff
@@ -0,0 +1,119 @@
+                  name="Status"
+                  placeholder="What's on your mind ?"
+                  value={this.state.Status}
+                  label="Status:&nbsp;&nbsp;&nbsp;&nbsp;"
```
You should use styling for the labels too to avoid the `&nbsp;`

Merdoth committed:
Merge pull request #24 from Merdoth/feature/154507643/delete-recipe

feature(delete-recipe and refractor code):

Merdoth committed:
Merge pull request #25 from Merdoth/Bug/travis-correction

Bug/travis correction

Merdoth committed:
refactor code

Merdoth committed:
Merge pull request #26 from Merdoth/feature/154611888/refactor-code-and-finishing-touches

refactor code

Merdoth committed:
feature(search-implementation):
- update controllers to accomodate changes made in the recipes controller
- change the repo_token in .coveralls.yml inorder to allow it build correctly
- remove searcha bar component cos there is an already existing one in the navigation bar
- update readme file
- add pagination to backend inorder to allow coordination of  results fetched from the database
- update actions and reducers to include the actions and reducers of new routes

[Delivers#154640968]

Merdoth committed:
Merge pull request #27 from Merdoth/feature/154640968/search-implementation

feature(search-implementation):

Merdoth committed:
bug(fix: code-refactor): - refactor code base
- fix code issues

Merdoth committed:
Merge pull request #28 from Merdoth/feature/154640968/search-implementation

bug(fix: code-refactor): - refactor code base

Merdoth committed:
Heroku setup

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
chore(Heroku and coveralls setup): run npm install to add dependencies
- update JSDoc
- update coveralls token
- modify package.json

[Delivers#154733060]

Merdoth committed:
Merge pull request #29 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #30 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #31 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #32 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #33 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #34 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #35 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' into chore/154733060/heroku-setup

Merdoth committed:
Merge pull request #36 from Merdoth/chore/154733060/heroku-setup

chore(Heroku and coveralls setup): run npm install to add dependencies

Merdoth committed:
Merge branch 'develop' of https://github.com/Merdoth/more-recipes into develop

Merdoth committed:
feature(votes): - did complete code refactor to make sure code is following clean code rules
-  modifiy votes controller

Merdoth committed:
feature(votes): - did complete code refactor to make sure code is following clean code rules
-  modifiy votes controller

Merdoth committed:
feature(votes): - did complete code refactor to make sure code is following clean code rules
-  modifiy votes controller

Merdoth committed:
Merge pull request #37 from Merdoth/feature/154840416/votes

Feature/154840416/votes

Merdoth committed:
Merge branch 'develop' into feature/154840416/votes

Merdoth committed:
Merge pull request #38 from Merdoth/feature/154840416/votes

feature(votes): - did complete code refactor to make sure code is fol…

Merdoth committed:
Merge branch 'develop' into feature/154840416/votes

Merdoth committed:
Merge pull request #39 from Merdoth/feature/154840416/votes

feature(votes): - did complete code refactor to make sure code is fol…

Merdoth committed:
feature(votes-favourites):  - update recipe reducers in components
- add not found page
- rearrange components
- link votes components actions and reducers
- link favourites components to actions and reducers

[Delivers#154840416]

Merdoth committed:
Merge pull request #40 from Merdoth/feature/154840416/votes-favourites

feature(votes-favourites):  - update recipe reducers in components

Merdoth committed:
chore(api-docs-to-app): -add api documentations  to application
- add access point for api-docs in app.js

[Delivers#154919994]

Merdoth committed:
Merge pull request #41 from Merdoth/chore/154919994/api-docs-to-app

chore(api-docs-to-app): -add api documentations  to application

Merdoth committed:
feature(ui and ux friendly): - modify validation inputs to suit users use
- display top recipes on home page

Merdoth committed:
feature(ui-ux friendly):  -update css styles to suit user satisfaction
- connect components with actions and reducers to enable  component access to store
- remove eslint flag on else statements
- modify validation for both client and server side

[Delivers#154929634]

Merdoth committed:
feature(ui-ux friendly):  -update css styles to suit user satisfaction
- connect components with actions and reducers to enable  component access to store
- remove eslint flag on else statements
- modify validation for both client and server side

[Delivers#154929634]

Merdoth committed:
Merge branches 'feature/154929634/ui-ux-friendly' and 'feature/154929634/ui-ux-friendly' of https://github.com/Merdoth/more-recipes into feature/154929634/ui-ux-friendly
Because an unwanted vaildations folder is present

Merdoth committed:
Merge branches 'feature/154929634/ui-ux-friendly' and 'feature/154929634/ui-ux-friendly' of https://github.com/Merdoth/more-recipes into feature/154929634/ui-ux-friendly
Because an unwanted vaildations folder is present

Merdoth committed:
Merge branches 'feature/154929634/ui-ux-friendly' and 'feature/154929634/ui-ux-friendly' of https://github.com/Merdoth/more-recipes into feature/154929634/ui-ux-friendly
Because an unwanted vaildations folder is present

Merdoth committed:
Merge pull request #42 from Merdoth/feature/154929634/ui-ux-friendly

Feature/154929634/ui ux friendly

Merdoth committed:
Merge pull request #42 from Merdoth/feature/154929634/ui-ux-friendly

Feature/154929634/ui ux friendly

Merdoth committed:
Merge pull request #43 from Merdoth/chore/154919994/api-docs-to-app

Chore/ api-docs

Merdoth committed:
Merge pull request #43 from Merdoth/chore/154919994/api-docs-to-app

Chore/ api-docs

Merdoth committed:
Merge pull request #44 from Merdoth/chore/154733060/heroku-setup

/chore/154919994/api-docs-to-app

Merdoth committed:
Merge pull request #45 from Merdoth/feature/154929634/ui-ux-friendly

feature/154929…

Merdoth committed:
Merge pull request #46 from Merdoth/feature/154929634/ui-ux-friendly

feature/154929634/ui-ux-friendly

Merdoth committed:
feature(ui/ux friendly): - update components
- update scss
- update server

Merdoth committed:
feature(users-test):  - fix  failing build process for coverage
- update package.json dependencies
- rename test files to differentiate them from  other folders with same name
- update  server controllers

Merdoth committed:
Merge branch 'develop' into feature/155103165/users-test

Merdoth committed:
Merge pull request #47 from Merdoth/feature/155103165/users-test

Feature/155103165/users test

Merdoth committed:
fix(env-example): - update env example to enable user easy access to run the app

Merdoth committed:
feature(get-user-favourites):
- add favourite component
-  update actions and routes to include favourites

[Delivers155263919]

Merdoth committed:
feature(get-user-favourites):
- add favourite component
-  update actions and routes to include favourites

[Delivers155263919]

Merdoth committed:
Merge pull request #48 from Merdoth/feature/155263919/get-user-favourites

feature(get-user-favourites):

Merdoth committed:
Merge pull request #49 from Merdoth/feature/155263919/get-user-favourites

feature(get-user-favourites):

Merdoth committed:
chore(jsdoc-fix): - update javascript documentation for all files that lack proper documentation

[Delivers155271552]

Merdoth committed:
chore(frontend-test-setup):
- install jest packages
- add jest test script to package.json file
- add frontend-test setup index file
- add mock local storage for frontend test

[Delivers#155285126]

Merdoth committed:
Merge pull request #51 from Merdoth/chore/155285126/frontend-test-setup

Chore/155285126/frontend test setup

Merdoth committed:
Merge pull request #51 from Merdoth/chore/155285126/frontend-test-setup

Chore/155285126/frontend test setup

Merdoth committed:
feature(frontend-tests):  - add reducser tests
- add actions auth tests

Merdoth committed:
Merge pull request #52 from Merdoth/feature/154929634/ui-ux-friendly

Merge pull request #51 from Merdoth/chore/155285126/frontend-test-setup

Merdoth committed:
Merge pull request #52 from Merdoth/feature/154929634/ui-ux-friendly

Merge pull request #51 from Merdoth/chore/155285126/frontend-test-setup

Merdoth committed:
chore(end2end tests): -add setup for nightwatch to package.json
- setup nightwatch config
- add e2e-test scripts to package.json

Merdoth committed:
feature(frontend-tests): -add test case for components
-refactor test cases in actions
-refactor test cases in reducers

[Delivers#155285075]

Merdoth committed:
Merge pull request #53 from Merdoth/feature/155285075/frontend-tests

Feature/155285075/frontend tests

Merdoth committed:
chore(end2end-test): -add end to end test for homepage
-add end to end test for user features
-update logo picture for api-docs

[Delivers#155413242]

Merdoth committed:
Merge pull request #54 from Merdoth/chore/155413242/end2end-test

chore(end2end-test): -add end to end test for homepage

Merdoth committed:
fix merge conflict

Merdoth committed:
feature(feedback fix): -update JSdoc  on every file
- cover more egdecases on my test suites

Merdoth committed:
feature(feedback-fix):  - fix controllers feedback
- run client test to update snapshot
- fix reducers feedback
- fix components  feedback

[Delivers #155432453]

Merdoth committed:
feature(feedback-fix):  - fix controllers feedback
- run client test to update snapshot
- fix reducers feedback
- fix components  feedback

[Delivers #155432453]

Merdoth committed:
fix merge conflict

Merdoth committed:
feature(feedback fix):
- update javascript documentation
- update api documentation
- implement upload input to display image name
- update test suites
- fix mobile responsiveness

[Delivers#155432453]

Merdoth committed:
feature(feedback-fix): - update token time out
- upfdate javascript documentation where necessary
- update back end test suites
- add assertions to e2e test

[Delivers#155432453]

Merdoth committed:
Merge pull request #57 from Merdoth/feature/155432453/feedback-fix

Feature/155432453/feedback fix
